commands/deploy.go

```diff
@@ -343,9 +343,41 @@ func (p *deploycmd) deployFuncV20180708(c *cli.Context, app *models.App, funcfil
 	if funcfile.Name == "" {
 		funcfile.Name = filepath.Base(filepath.Dir(funcfilePath)) // todo: should probably make a copy of ff before changing it
 	}
-	fmt.Printf("Deploying %s to app: %s\n", funcfile.Name, app.Name)
 
-	var err error
+	oracleProvider, _ := getOracleProvider()
+	if oracleProvider != nil && oracleProvider.ImageCompartmentID != "" {
+		// If the provider is Oracle and ImageCompartmentID is present, we need to deploy image to the ImageCompartmentID.
+		// The repository name should be unique throughout a tenancy. We check if a repository exists in the compartment and create it if it doesn't already exist.
+		// If the creation fails, it could be because the repository name aready exists in a different compartment.
+
+		if client.Registry() == "" {
+			return errors.New("registry name must be set, be sure to configure your context file")
+		}
+		res, err := getRepositoryNamePrefix(client.Registry())
```

Comment:
I think that the existing function `getRepositoryName` will do what you want here. Additionally it'll support `name` containing a different registry
```suggestion
		res, err := getRepositoryName(funcfile)
```